### DISCUSSION TITLE
HTTP(S) + JSON Abfrage Stromzähler nur in kW möglich - Umrechnung in W über OpenDTU-OnBattery Weboberfläche möglich?

### DISCUSSION THREAD

**Discussion by the-lonely-one** [hoylabs/OpenDTU-OnBattery#882] [Q&A] [answered]
Hallo OpenDTU-OnBattery-Gemeinde,

habe bei mir ein Hutschienen-Messgerät in die Netzeinspeisung eingebaut, bei dem ich den Wert der Gesamt-Wirkleistung per HTTP + JSON abfragen kann!

Die URL und der JSON Pfad ist unter "Stromzähler Einstellungen" soweit eingerichtet und ich bekomme auch den richtigen Wert abgefragt, aber leider nur in der Einheit "kW".
Die OpenDTU-OnBattery bzw. der DPL benötigt diesen Wert aber in "W"
Am Hutschienen-Messgerät selber kann die Ausgabe für diesen Wert nicht in "W" umgestellt werden!

Daher die Frage:
Gibt es über die Weboberfläche von OpenDTU-OnBattery irgendwie die Möglichkeit diesen Wert bei der HTTP + JSON-Abfrage so zu manipuliere (z.B. Wert x 1000), das ich den Wert in "W" an die OpenDTU-OnBattery übergeben kann?

Aktuell habe ich mir eine "Brücke" über Node-RED (läuft auf meinem 24/7-Server) gebaut!

Das heißt:
Node-RED fragt alle 5 Sekunden den Wert per HTTP + JSON vom Hutschienen-Messgerät ab.
Anschließen wird dieser Wert mit 1000 multipliziert (kW->W) und dann an meinen MQTT-Broker geschickt!
Die OpenDTU-OnBattery holt sich dann diesen Wert per MQTT ab!

Da ich aber eine direkte Verbindung zwischen dem Hutschienen-Messgerät und der OpenDTU-OnBattery bevorzugen würde, 
hoffe ich nun, das ihr mir da weiter helfen könnt!

**Comment by schlimmchen** [comment]
Siehe #451.

Also nein, es geht zur Zeit nicht, aber es wäre eine sinnvolle Einstellung.

**Reply by the-lonely-one** [reply]
Danke für die Info!
Ich war wohl nicht der Einzige, der so eine Funktion benötigt hätte!😊
Werde dann erst mal mit meiner „Brücke“ so weiter leben!
Ist zwar nicht so optimal, da für die Übermittlung der Wirkleistung vom Messgerät zur OpenDTU-OnBattery aktuell auch immer NodeRED benötigt wird, aber es funktioniert!

Und bevor ich es vergesse!
Ich finde das Projekt: „OpenDTU-OnBattery“ immer noch genial und danke allen Beteiligten, die so viel Zeit und Energie in dieses Projekt stecken um den Funktionsumfang zu erweitern und die Funktionen zu verbessern!
Kann man nicht oft genug sagen!

**Comment by schlimmchen** [comment, marked as answer]
Siehe #887.
